web/src/app/donate/DonateComponent.js

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  FormControl,
+  FormGroup,
+  ControlLabel,
+  InputGroup,
+  Grid,
+  Row,
+  Col
+} from 'react-bootstrap';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import PayWithPayPalComponent from './PayWithPayPalComponent';
+import styles from './DonateStyles.scss';
+
+class DonateComponent extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      amount: '',
+      bounty: '',
+      nameError: null,
+      amountError: null,
+      bountyError: null
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+    this.validateAmount = this.validateAmount.bind(this);
+  }
+
+  submitForm(e) {
+    e.preventDefault();
+    this.validateName();
+    this.validateAmount();
+    this.validateBounty();
+
+    if ((this.state.nameError && this.state.nameError.type === null) &&
+        (this.state.amountError && this.state.amountError.type === null) &&
+        (this.state.bountyError && this.state.bountyError.type === null)) {
+          // everything is valid
```

Comment:
nit: consider removing the comment and wrapping these lines in a well named function